cpp/src/parquet/page_index_benchmark.cc

```diff
@@ -0,0 +1,153 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <vector>
+
+#include "benchmark/benchmark.h"
+
+#include "parquet/metadata.h"
+#include "parquet/page_index.h"
+#include "parquet/schema.h"
+#include "parquet/test_util.h"
+#include "parquet/thrift_internal.h"
+
+namespace parquet {
+
+void BM_ReadOffsetIndex(::benchmark::State& state) {
+  auto builder = OffsetIndexBuilder::Make();
+  const size_t num_pages = state.range(0);
+  constexpr size_t page_size = 1024;
+  constexpr size_t first_row_index = 10000;
+  for (size_t i = 0; i < num_pages; ++i) {
+    builder->AddPage(page_size * i, page_size, first_row_index * i);
+  }
+  const int64_t final_position = 4096;
+  builder->Finish(final_position);
+  auto sink = CreateOutputStream();
```

Comment:
This benchmark reads from memory only, so it mainly measures the time spent on deserialization. We can add a benchmark (like s3fs_benchmark.cc) in the future to read from cloud object store and test the capability of PageIndexReader to combine small I/Os from different columns.

Reply:
Instead of depending on a fluctuating external source, you can use `SlowInputStream` or `SlowRandomAccessFile`.

Reply:
I think handling benchmark on slow input is a bit complex, let me implement it later

Reply:
Yes, I don't think it's important for this PR (or perhaps even in general).